### DISCUSSION TITLE
Connect to a UNIX socket through a file descriptor

### DISCUSSION THREAD

**Discussion by davidbrochart** [agronholm/anyio#682] [Q&A]
I'd like to [use AnyIO with UNIX sockets](https://anyio.readthedocs.io/en/stable/networking.html#working-with-unix-sockets). It looks like the API only accepts a file system path to connect to a socket, but in my case I already have a file descriptor (returned by [pty.fork()](https://docs.python.org/3/library/pty.html#pty.fork)).
Is there a way to do that?

**Comment by agronholm** [comment]
Can you do that with asyncio?

**Reply by davidbrochart** [reply]
Yes, with [loop.add_reader(fd, callback, *args)](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.add_reader).

**Reply by agronholm** [reply]
I'm willing to accept a PR (accompanied with the appropriate docs/test updates of course).

**Reply by davidbrochart** [reply]
Actually it looks like I can solve my problem using [streams](https://anyio.readthedocs.io/en/stable/streams.html).
